internal/lib/stats/network_metrics.go

```diff
@@ -0,0 +1,134 @@
+package statsserver
+
+import (
+	"github.com/cri-o/cri-o/internal/lib/sandbox"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+	types "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func (ss *StatsServer) GenerateNetworkMetrics(sb *sandbox.Sandbox) []*types.Metric {
+	var metrics []*types.Metric
+
+	links, err := netlink.LinkList()
+	if err != nil {
+		logrus.Errorf("Unable to retrieve network namespace links %s: %v", sb.ID(), err)
+		return nil
+	}
+	if len(links) == 0 {
+		logrus.Infof("Network links are not available.")
+		return nil
+	}
+	for i := range links {
+		attrs := links[i].Attrs()
+		if attrs != nil {
```

Comment:
Consistency in style.

```suggestion
		if attrs := links[i].Attrs(); attrs != nil {
```